test/univariate/continuous/triangular.jl

```diff
@@ -0,0 +1,473 @@
+    t == "Int16" && return convert(Int16, x)
+    t == "Int32" && return convert(Int32, x)
+    t == "Int64" && return convert(Int64, x)
+    t == "Int128" && return convert(Int128, x)
+    t == "BigInt" && return convert(BigInt, x)
+    t == "UInt8" && return convert(UInt8, x)
+    t == "UInt16" && return convert(UInt16, x)
+    t == "UInt32" && return convert(UInt32, x)
+    t == "UInt64" && return convert(UInt64, x)
+    t == "UInt128" && return convert(UInt128, x)
+    # t == "Bool" && return convert(Bool, x)
+end
+
+types = ["Float16", "Float32", "Float64", "BigFloat", "Rational"
+        , "Int8", "Int16", "Int32", "Int64", "Int128", "BigInt"
+        , "UInt8", "UInt16", "UInt32", "UInt64", "UInt128"] #Bool
+n = length(types)^2
+typeslist = Array{String}(undef, n, 2)
+j = 1
+for t in eachindex(types)
+    for u in eachindex(types)
+        typeslist[j, 1] = types[t]
+        typeslist[j, 2] = types[u]
+        global j += 1
+    end
+end
+
+@testset "triangular" begin
+    
+    @test_throws ArgumentError TriangularDist(1, 0, 0)
+    @test_throws ArgumentError TriangularDist(1, 1, 0)
+    @test_throws ArgumentError TriangularDist(0, 0, 1)
+    @test_throws ArgumentError TriangularDist(0, 0, 0)
+
+    @testset "special_triangle" begin
+        a = 0
+        b = 1
+        c = 1
+        xa = a
+        xb = b
+        xlow = a - ϵ
+        xhigh = b + ϵ
+        xac = middle(a, c)
+        xbc = middle(b, c)
+
+        d = TriangularDist(a, b, c)
+        @test mean(d) == 2 / 3
+        @test var(d) == 1 / 18
+        @test pdf(d, xa) == 2 * xa
+        @test pdf(d, xb) == 2 * xb
+        @test pdf(d, xac) == 2 * xac
+        @test pdf(d, xbc) == 2 * xbc
+        @test pdf(d, xlow) == 0
+        @test pdf(d, xhigh) == 0
+        @test cdf(d, xa) == xa^2
+        @test cdf(d, xb) == xb^2
+        @test cdf(d, xac) == xac^2
+        @test cdf(d, xbc) == xbc^2
+        @test cdf(d, xlow) == 0
+        @test cdf(d, xhigh) == 1
+    end
+
+    @testset "regular_triangle" begin
+        a = 1
+        b = 5
+        c = 4
+        xa = a
+        xb = b
+        xlow = a - ϵ
+        xhigh = b + ϵ
+        xac = middle(a, c)
+        xbc = middle(b, c)
+        t1 = 1
+        t2 = 2
+
+        d = TriangularDist(a, b, c)
+        @test params(d) == (a, b, c)
+        @test mode(d) == c
+        @test mean(d) == (a + b + c) / 3
+        @test median(d) == (c >= middle(a, b) ?
+            a + sqrt((b - a) * (c - a) / 2) :
+            b - sqrt((b - a) * (b - c) / 2))
+        @test var(d) == (a^2 + b^2 + c^2 - a * b - a * c - b * c) / 18
+        @test skewness(d) == sqrt(2) * (a + b - 2c) * (2a - b - c) * (a - 2b + c) /
+            (5 * (a^2 + b^2 + c^2 - a * b - a * c - b * c) ^ (3 / 2))
+        @test kurtosis(d) == -3 / 5
+        @test entropy(d) == 1 / 2 + log((b - a) / 2)
+        @test pdf(d, xa) == 2 * (xa - a) / ((b - a) * (c - a))
+        @test pdf(d, xb) == 2 * (b - xb) / ((b - a) * (b - c))
```

Comment:
Can these tests be unified? It seems a bit redundant to test every scenario completely separately.